### PR TITLE
Added some Docblocks

### PR DESCRIPTION
**Pull request by vanbrabantf**
### Added
Added some docblock suggar to the methodes

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -136,6 +162,11 @@ public function getConfig($option = null)
             : (isset($this->config[$option]) ? $this->config[$option] : null);
     }
 
+    /**
+     * @param $uri
+     * @param array $config
+     * @return Psr7\Uri|UriInterface
```
Comment (hannesvdvreken) [inline src/Client.php]: I'd stick to `UriInterface`

File: src/Handler/MockHandler.php [hunk 2]
```diff
@@ -149,6 +154,12 @@ public function count()
         return count($this->queue);
     }
 
+    /**
+     * @param RequestInterface $request
+     * @param array $options
+     * @param ResponseInterface|null $response
```
Comment (hannesvdvreken) [inline src/Handler/MockHandler.php]: I'd leave off the `|null`

File: src/Handler/StreamHandler.php [hunk 3]
```diff
@@ -127,6 +141,11 @@ private function createResponse(
         return new FulfilledPromise($response);
     }
 
+    /**
+     * @param StreamInterface $stream
+     * @param array $options
+     * @return Psr7\Stream|StreamInterface
```
Comment (hannesvdvreken) [inline src/Handler/StreamHandler.php]: Same here: `StreamInterface` is all the requester needs to know.

File: src/RedirectMiddleware.php [hunk 4]
```diff
@@ -115,6 +115,11 @@ public function checkRedirect(
         return $promise;
     }
 
+    /**
+     * @param PromiseInterface $promise
+     * @param $uri
+     * @return mixed
```
Comment (hannesvdvreken) [inline src/RedirectMiddleware.php]: $promise->then will always return `PromiseInterface` see https://github.com/guzzle/promises/blob/master/src/PromiseInterface.php#L26-L28

File: src/Handler/MockHandler.php [hunk 5]
```diff
@@ -59,6 +59,11 @@ public function __construct(
         }
     }
 
+    /**
+     * @param RequestInterface $request
+     * @param array $options
+     * @return mixed
```
Comment (hannesvdvreken) [inline src/Handler/MockHandler.php]: afaik middlewares return objects which implement `ResponseInterface`

File: src/RetryMiddleware.php [hunk 6]
```diff
@@ -102,6 +112,11 @@ private function onRejected(RequestInterface $req, array $options)
         };
     }
 
+    /**
+     * @param RequestInterface $request
+     * @param array $options
+     * @return mixed
```
Comment (hannesvdvreken) [inline src/RetryMiddleware.php]: same here: `@return \Psr\Http\Message\RequestInterface` (or import it).

File: src/UriTemplate.php [hunk 7]
```diff
@@ -35,6 +35,11 @@ class UriTemplate
         '%40', '%21', '%24', '%26', '%27', '%28', '%29', '%2A', '%2B', '%2C',
         '%3B', '%3D');
 
+    /**
+     * @param $template
+     * @param array $variables
+     * @return mixed
+     */
```
Comment (hannesvdvreken) [inline src/UriTemplate.php]:
```php
/**
 * @param string $template
 * @param array $variables
 *
 * @return string
 */
```

**Comment by hannesvdvreken** [comment]
Try not to leave the return type undefined. `@return mixed` is not very useful to anyone. I explained one use case in this blog post: http://blog.madewithlove.be/post/return-types/

Thanks for the docblocks! Super useful for my IDE to help me use these methods without having to dive into the source.

**Comment by mtdowling** [comment]
The Guzzle project does not place docblocks on methods that are documented by their interface. Newer versions of PHPStorm and possibly other PHP IDE's now properly show interface documentation on implementations that have no docblock.

Any of the docblocks you've added that are not covered by an interface or parent's class docblock are good additions though. I'd be happy to merge a PR that adds those kinds of missing docblocks.

**Comment by hannesvdvreken** [comment]
So, not even a docblock like this:

```php
/**
 * {@inherit}
 */
```

Ok :-) @vanbrabantf this is project specific advice from the maintainer. Too bad I can't find that guideline in a contributing.md file. There's still a good chance to contribute here! Go for it.

**Comment by mtdowling** [comment]
That's a good point. I need to add a contributing.md file with this called out.

**Comment by vanbrabantf** [comment]
Should I include the change to the "contributing.md" in my next commit? No problem, I'll just use: 

"The Guzzle project does not place docblocks on methods that are documented by their interface. Newer versions of PHPStorm and possibly other PHP IDE's now properly show interface documentation on implementations that have no docblock."

**Comment by hannesvdvreken** [comment]
@vanbrabantf Maybe in a separate PR. There is no contributing.md file yet. It will need to contain more than that sentence.

**Comment by vanbrabantf** [comment]
I updated the PR with only the methodes that are not covered with the interface. These are mostly private functions so I'm not sure how useful those are.

**Comment by mtdowling** [comment]
Thanks for the modifications. I think the docblocks that you added to private methods are unnecessary -- they don't add anything over the method signature. Can you remove those docblocks?

**Comment by mtdowling** [comment]
If we're going to add docs, we shouldn't just add them for the sake of having a docblock. Let's actually document the parameters. If we aren't documenting the parameters, then adding a docblock doesn't provide any value because everything can be inferred already by an IDE.

**Comment by jfountain** [comment]
In regards to contributions guidelines, this is a good place to start...

http://todogroup.org/opencodeofconduct/

File: src/RetryMiddleware.php [hunk 8]
```diff
@@ -101,7 +101,7 @@ private function onRejected(RequestInterface $req, array $options)
             return $this->doRetry($req, $options);
         };
     }
-
+
```
Comment (mtdowling) [inline src/RetryMiddleware.php]: extra spaces

File: src/Handler/EasyHandle.php [hunk 9]
```diff
@@ -77,6 +77,9 @@ public function createResponse()
         );
     }
 
+    /**
+     * @param string $name
```
Comment (mtdowling) [inline src/Handler/EasyHandle.php]: I think this docblock is unnecessary

File: src/Handler/CurlMultiHandler.php [hunk 10]
```diff
@@ -41,6 +41,10 @@ public function __construct(array $options = [])
             ? $options['select_timeout'] : 1;
     }
 
+    /**
+     * @param $name
+     * @return resource
+     */
     public function __get($name)
```
Comment (mtdowling) [inline src/Handler/CurlMultiHandler.php]: I think this docblock is unnecessary

File: src/Cookie/SetCookie.php [hunk 11]
```diff
@@ -82,6 +82,9 @@ public function __construct(array $data = [])
         }
     }
 
+    /**
+     * @return string
+     */
     public function __toString()
```
Comment (mtdowling) [inline src/Cookie/SetCookie.php]: I think this docblock is unnecessary

File: src/Handler/CurlMultiHandler.php (hunk 10 above)
Comment (GrahamCampbell) [inline src/Handler/CurlMultiHandler.php]: This docblock is also incorrectly formatted. The type is always required.

**Comment by mtdowling** [comment]
I'll go ahead and close this. I think there are several additions here that aren't necessary. There are a couple good docblocks, but I think those could be submitted in a fresh PR.

**Pull request closed without merging**